src/figuresClasses.ts

```diff
@@ -1,11 +1,93 @@
-export interface Figure {}
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+type Color = 'red' | 'green' | 'blue';
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public aSide: number,
+    public bSide: number,
+    public cSide: number,
+  ) {
```

Comment:
[CODE KNOWLEDGE] - Since TypeScript implicitly creates class properties from constructor parameters with access modifiers, you don't need to manually assign them. You can remove the assignments from the constructor in the `Triangle` class.